int8 markerID
---
bool hasMask